frontend/src/store/slices/UserSlice.ts

```diff
@@ -0,0 +1,27 @@
+import { createSlice } from "@reduxjs/toolkit";
+import type { RootState } from "store/store";
+
+export interface User {
+    username: string;
+}
+
+const initialState: User = {
+    username: "Dohn Joe"
+};
+
+export const userSlice = createSlice({
+    name: "user",
+    initialState,
+    reducers: {
+        editUsername: (state, action) => {
```

Comment:
that's should be a async because u want to implement async logic from backend.

But that's good think if u want to implement dammy logic now

Reply:
done